lib/diagnostic.js

```diff
@@ -39,20 +51,39 @@ const owner = null
 // Methods: vocalize
 
 // Replace null with your answer
-const pet = null
+const pet = {
+  name: 'Buster',
+  species: 'cat',
+  breed: 'unknown',
+  vocalization: 'meows',
+  vocalize: function() { return console.log(this.vocalization)}
+}
+//console.log(pet)
+//console.log(pet.vocalize())
 
 // 4. Create a constructor function for pet objects, as described above. The
 // constructor function should be responsible for setting the values of `name`,
 // `species`, `breed`, and `vocalization`.
 
-const Pet = function () {
-  /* Your answer here */
+const Pet = function (petName, petSpecies, petBreed, petVocalization) {
+ this.name = petName
+ this.species = petSpecies
+ this.breed = petBreed
+ this.vocalization = petVocalization
 }
+//let meow = new Pet('m', 's', 's', 't')
+//console.log(meow)
 
 // 5. Attach a `vocalize` method to the Pet's prototype.  The `vocalize` method
 // should return the pet's vocalization attribute.
 
 /* your answer here */
+Pet.prototype.vocalize = function () {
+  const vocalize = this.vocalization
```

Comment:
You don't actually have to specify this variable inside here, the function will know what `vocalization` is